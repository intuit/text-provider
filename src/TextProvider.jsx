import React from 'react';
import PropTypes from 'prop-types';

export const TextContext = React.createContext({});

const TextProvider = (props) => {
  const { globalText, children } = props;
  return <TextContext.Provider value={globalText}>{children}</TextContext.Provider>;
};

TextProvider.propTypes = {
  globalText: PropTypes.objectOf(PropTypes.string),
  children: PropTypes.node.isRequired,
};

TextProvider.defaultProps = {
  globalText: {},
};

export { TextProvider as default };
