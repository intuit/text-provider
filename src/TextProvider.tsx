import React from 'react';

export const TextContext: React.Context<{}> = React.createContext({});

/**
 * Definition
 */
interface TextProviderProps {

/**
 * Key-value pair of text strings
 */
    globalText?: { [key: string]: string};

    children: any;
}

const TextProvider: React.FC<TextProviderProps> = (props: TextProviderProps) => {
  const { globalText, children } = props;
  return (
    <TextContext.Provider value={globalText}>
      {children}
    </TextContext.Provider>
  );
};

/**
 * Set default props
 */
TextProvider.defaultProps = {
  globalText: {},
};

export { TextProvider as default };
