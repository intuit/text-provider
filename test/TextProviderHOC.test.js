import React from 'react';
import renderer from 'react-test-renderer';
import TextProvider from '../src/TextProvider';
import withTextProvider from '../src/TextProviderHOC';

describe('<TextProviderHOC /> rendering', () => {
  class SomeComponent extends React.Component {
    constructor(props) {
      super(props);
    }

    render() {
      return <div>{this.props.id ? this.props.id : 'id'}</div>;
    }
  }

  it('should render TextProviderHOC correctly', () => {
    const MockWithHOC = withTextProvider(SomeComponent);

    const wrapper = renderer.create(<MockWithHOC />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should add getTextForKey as prop for component', () => {
    const MockWithHOC = withTextProvider(SomeComponent);

    const wrapper = renderer.create(<MockWithHOC />).root;
    expect(wrapper.findByType(SomeComponent).props.getTextForKey).toBeDefined();

    // TextProviderHOC uses {} as default context
    expect(wrapper.findByType(SomeComponent).props.getTextForKey('test')).toEqual('');
  });

  it('should render the component provided along with props', () => {
    const MockWithHOC = withTextProvider(SomeComponent);
    const wrapper = renderer.create(<MockWithHOC id="id" />).root;
    expect(wrapper.findByType(SomeComponent).props.id).toEqual('id');
  });

  it('test getTextForKey with context', () => {
    let context = { id: 'test id' };
    const MockWithHOC = withTextProvider(SomeComponent);
    let wrapper = renderer.create(<TextProvider globalText={context} children={<MockWithHOC />} />)
      .root;
    expect(wrapper.findByType(SomeComponent).props.getTextForKey('id')).toEqual(context.id);
  });
});
