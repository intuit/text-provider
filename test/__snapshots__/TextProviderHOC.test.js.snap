// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextProviderHOC /> rendering should render TextProviderHOC correctly 1`] = `
<div>
  id
</div>
`;
